sdk/src/metrics/metric_reader.cc

```diff
@@ -46,18 +47,22 @@ bool MetricReader::Collect(nostd::function_ref<bool(MetricData)> callback) noexc
 bool MetricReader::Shutdown(std::chrono::microseconds timeout) noexcept
 {
   bool status = true;
-
   if (IsShutdown())
   {
     OTEL_INTERNAL_LOG_WARN("MetricReader::Shutdown - Cannot invoke shutdown twice!");
   }
+
+  {
+    const std::lock_guard<opentelemetry::common::SpinLockMutex> locked(lock_);
+    shutdown_ = true;
+  }
+
   if (!OnShutDown(timeout))
   {
     status = false;
     OTEL_INTERNAL_LOG_WARN("MetricReader::OnShutDown Shutdown failed. Will not be tried again!");
   }
   const std::lock_guard<opentelemetry::common::SpinLockMutex> locked(lock_);
```

Comment:
`status` is local to the function.

Reply:
Yes, it is meant to be local. Do you see any issue here?

Reply:
Sorry I meant we don't need the lock in line 65.